nodes/session.go

```diff
@@ -33,18 +31,21 @@ func Session(ctx context.Context, ns []p2plab.Node, fn func(context.Context) err
 	defer span.Finish()
 	sctx := opentracing.ContextWithSpan(ctx, span)
 
+	var ids []string
+	for _, n := range ns {
+		ids = append(ids, n.ID())
+	}
+
 	eg, gctx := errgroup.WithContext(sctx)
 
 	zerolog.Ctx(ctx).Info().Msg("Starting a session for benchmarking")
 	go logutil.Elapsed(gctx, 20*time.Second, "Starting a session for benchmarking")
-
-	cancels := make([]context.CancelFunc, len(ns))
-	for i, n := range ns {
-		i, n := i, n
+	/*var cancels = make([]context.CancelFunc, 0, len(ns))
```

Comment:
Can you add a comment here so the context isn't lost?

Reply:
That is definitely a good idea